File: apps/java/src/main/java/com/functions/ScheduledCleanupFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;

/**
 * Timer trigger — scheduled cleanup job running at 2:00 AM UTC daily.
 */
public class ScheduledCleanupFunction {

    @FunctionName("scheduledCleanup")
    public void run(
            @TimerTrigger(
                name = "timer",
                schedule = "0 0 2 * * *")
            String timerInfo,
            final ExecutionContext context) {

        context.getLogger().info("Scheduled cleanup executed at: " + java.time.Instant.now());
        context.getLogger().info("Timer info: " + timerInfo);
    }
}

File: apps/dotnet/Functions/ScheduledCleanupFunction.cs
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Scheduled cleanup — runs daily at 2:00 AM UTC.
/// Binding: Timer trigger with CRON expression "0 0 2 * * *".
/// </summary>
public class ScheduledCleanupFunction
{
    private readonly ILogger<ScheduledCleanupFunction> _logger;

    public ScheduledCleanupFunction(ILogger<ScheduledCleanupFunction> logger)
    {
        _logger = logger;
    }

    [Function("scheduledCleanup")]
    public void Run(
        [TimerTrigger("0 0 2 * * *")] TimerInfo timer)
    {
        _logger.LogInformation("Scheduled cleanup fired at {Timestamp}", DateTime.UtcNow.ToString("o"));

        if (timer.IsPastDue)
        {
            _logger.LogWarning("Timer trigger is past due");
        }
    }
}
